src/lib.rs

```diff
@@ -728,6 +862,34 @@ mod tests {
         }
     }
 
+    /// Check that (sequentially) writing to a triple buffer works
+    #[test]
+    fn sequential_guarded_write() {
+        // Let's create a triple buffer
+        let mut buf = TripleBuffer::new(&false);
+
+        // Back up the initial buffer state
+        let old_buf = buf.clone();
+
+        // Perform a write
+        *buf.input.input_buffer_publisher() = true;
+
+        // Check new implementation state
+        {
+            // Starting from the old buffer state...
+            let mut expected_buf = old_buf.clone();
+
+            // ...write the new value in and swap...
+            {
+                *expected_buf.input.input_buffer_publisher() = true;
+            }
```

Comment:
The intent of the original `sequential_write` test was to check that doing a write using the high-level `write()` method and using a low-level `input_buffer()`/`publish()` pair results in the exact same final state.

To replicate this logic here, you should do...

```suggestion
        // Perform a write
        *buf.input.input_buffer_publisher() = true;

        // Check new implementation state
        {
            // Starting from the old buffer state...
            let mut expected_buf = old_buf.clone();

            // ...write the new value in and swap...
            *expected_buf.input.input_buffer() = true;
            expected_buf.input.publish();
```